identity-verification/module_2/2.1 Rekognition IDV API Examples.ipynb

```diff
@@ -0,0 +1,706 @@
+    "print(\"-- ExternalImageID : {} \\n\".format(external_image_id))\n",
+    "\n",
+    "print(json.dumps(response, indent=3))  \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## SearchFaces \n",
+    "-----\n",
+    "For a given input **face ID**, search the collection for  matching faces. You get a face ID when you add a face to the collection using the IndexFaces operation(above) we are simply going to index additional faces below, and snag a couple of face ID. The operation compares the features of the input face with faces in the specified collection. Essentially matching faces within a collection. \n",
+    "\n",
+    "\n",
+    "\n",
+    "<div class=\"alert alert-info\"><strong> Setup </strong>\n",
+    "\n",
+    "- Index additional faces, in this case we'll use our image_map dataframe and take the first 10 selfie_images \n",
+    "- Make note of a face ID returned from the index \n",
+    "- SearchFaces for a given *FaceID* and assess the match. \n",
+    "\n",
+    "</div>\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "## Index several faces \n",
+    "dict_of_faces = image_map[[\"reference_name\",\"selfie_0\"]].head(10).to_dict('records')\n",
+    "\n",
+    "for rec in dict_of_faces:\n",
+    "  \n",
+    "    response = rek_client.index_faces(\n",
+    "        CollectionId= collection_name,\n",
+    "        Image={\n",
+    "            'S3Object': {\n",
+    "                'Bucket': bucket_name,\n",
+    "                'Name': rec[\"selfie_0\"],\n",
+    "            }\n",
+    "        },\n",
+    "        ExternalImageId=rec['reference_name'],\n",
+    "        DetectionAttributes=[\n",
+    "            'DEFAULT',\n",
+    "        ],\n",
+    "        MaxFaces=1, # maximum faces detected \n",
+    "        QualityFilter='AUTO' # apply the quality filter. \n",
+    "        )\n",
+    "    face_id = response['FaceRecords'][0]['Face']['FaceId']\n",
+    "    print(\"ImageName: {}, FaceID: {}\".format(rec[\"selfie_0\"], face_id))\n",
+    "    \n",
+    "\n",
+    "print(\"indexing complete\")\n",
+    "    \n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#####  Snag one of the FaceIDs above \n",
+    "----\n",
+    "Pass the FaceID above to search_faces, this will identify potential matches within the collection. \n",
+    "\n",
+    "**Select a face id from the results above and enter it below**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "\n",
+    "face_id = \"\" # replace with your own face ID\n",
+    "try:\n",
+    "    response = rek_client.search_faces(\n",
+    "        CollectionId=collection_name,\n",
+    "        FaceId=face_id,\n",
+    "        MaxFaces=10\n",
+    "    )\n",
+    "    print(json.dumps(response, indent=3))  \n",
+    "except:\n",
```

Comment:
This should be scoped to Rekognition.Client.exceptions.ResourceNotFoundException